[Site Isolation] Begin filling out and connecting RemoteMediaSessionManager and RemoteMediaSessionManagerProxy

eric-carlson opened the pull request:
#### fa789dd834b7818cff1cf2ee9ca080aa122c887e
<pre>
[Site Isolation] Begin filling out and connecting RemoteMediaSessionManager and RemoteMediaSessionManagerProxy
<a href="https://bugs.webkit.org/show_bug.cgi?id=302546">https://bugs.webkit.org/show_bug.cgi?id=302546</a>
<a href="https://rdar.apple.com/164742719">rdar://164742719</a>

Reviewed by Jean Yves Avenard.

Beging filling out the implementations of RemoteMediaSessionManager and
RemoteMediaSessionManagerProxy.

No new tests, the remote session manager is disabled by default and is not testable yet.

* Source/WebCore/platform/audio/MediaSessionManagerInterface.cpp:
(WebCore::MediaSessionManagerInterface::addSession):
* Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
(WebCore::MediaSessionManagerInterface::logger const):
* Source/WebCore/platform/audio/PlatformMediaSession.h:
(WebCore::PlatformMediaSession::create): Deleted.
(WebCore::PlatformMediaSession::PlatformMediaSession): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
(WebCore::PlatformMediaSessionInterface::setMediaSessionIdentifier):
(WebCore::PlatformMediaSessionInterface::client const):
(WebCore::PlatformMediaSessionInterface::checkedClient const):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::sessions const):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::updateSessionState):
(WebCore::MediaSessionManagerCocoa::scheduleSessionStatusUpdate):
(WebCore::MediaSessionManagerCocoa::addSession):
(WebCore::MediaSessionManagerCocoa::removeSession):
(WebCore::MediaSessionManagerCocoa::setCurrentSession):
(WebCore::MediaSessionManagerCocoa::addSupportedCommand):
(WebCore::MediaSessionManagerCocoa::removeSupportedCommand):
(WebCore::MediaSessionManagerCocoa::supportedCommands const):
(WebCore::MediaSessionManagerCocoa::updateNowPlayingInfo):
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
(WebCore::MediaSessionManageriOS::logClassName const):
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.cpp:
(WebKit::RemoteMediaSessionClientProxy::RemoteMediaSessionClientProxy):
(WebKit::RemoteMediaSessionClientProxy::resumeAutoplaying):
(WebKit::RemoteMediaSessionClientProxy::mayResumePlayback):
(WebKit::RemoteMediaSessionClientProxy::suspendPlayback):
(WebKit::RemoteMediaSessionClientProxy::didReceiveRemoteControlCommand):
(WebKit::RemoteMediaSessionClientProxy::shouldOverrideBackgroundPlaybackRestriction const):
(WebKit::RemoteMediaSessionClientProxy::setShouldPlayToPlaybackTarget):
(WebKit::RemoteMediaSessionClientProxy::sessionManager const):
(WebKit::RemoteMediaSessionClientProxy): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.h:
(WebKit::RemoteMediaSessionClientProxy::create): Deleted.
(WebKit::RemoteMediaSessionClientProxy::selectBestMediaSession): Deleted.
(WebKit::RemoteMediaSessionClientProxy::protectedLogger const): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.cpp:
(WebKit::RemoteMediaSessionManagerProxy::RemoteMediaSessionManagerProxy):
(WebKit::RemoteMediaSessionManagerProxy::addMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::removeMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::setCurrentMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::updateMediaSessionState):
(WebKit::RemoteMediaSessionManagerProxy::findSession):
(WebKit::RemoteMediaSessionManagerProxy::addSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::removeSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::setCurrentSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::updateSessionState): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.h:
(WebKit::RemoteMediaSessionManagerProxy::process const):
(WebKit::RemoteMediaSessionManagerProxy::ref const): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::deref const): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp:
(WebKit::RemoteMediaSessionProxy::RemoteMediaSessionProxy):
(WebKit::RemoteMediaSessionProxy::~RemoteMediaSessionProxy):
(WebKit::RemoteMediaSessionProxy::updateState):
(WebKit::RemoteMediaSessionProxy::setShouldPlayToPlaybackTarget):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h: Added.
* Source/WebKit/UIProcess/XR/PlatformXRSystem.h:
* Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp:
(WebKit::SystemSettingsManagerProxy::updateFontProperties):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.cpp:
(WebKit::RemoteMediaSessionManager::addSession):
(WebKit::RemoteMediaSessionManager::removeSession):
(WebKit::RemoteMediaSessionManager::setCurrentSession):
(WebKit::RemoteMediaSessionManager::updateSessionState):
(WebKit::RemoteMediaSessionManager::sessionWithIdentifier):
(WebKit::RemoteMediaSessionManager::clientShouldResumeAutoplaying):
(WebKit::RemoteMediaSessionManager::clientMayResumePlayback):
(WebKit::RemoteMediaSessionManager::clientShouldSuspendPlayback):
(WebKit::RemoteMediaSessionManager::clientSetShouldPlayToPlaybackTarget):
(WebKit::RemoteMediaSessionManager::clientDidReceiveRemoteControlCommand):
(WebKit::RemoteMediaSessionManager::currentSessionState):
(WebKit::RemoteMediaSessionManager::updateCachedSessionState):
(WebKit::RemoteMediaSessionManager::fullSessionState):
(WebKit::RemoteMediaSessionManager::sharedPreferencesForWebProcess const):
(WebKit::platformSessionState): Deleted.
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.h:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.messages.in:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.h:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.serialization.in:

Canonical link: <a href="https://commits.webkit.org/303123@main">https://commits.webkit.org/303123@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c82e62fdd01d5bc6bfb83861651dbddaeb27977a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131421 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3752 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42385 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138916 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/83187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3a350235-e790-4a87-98b0-62329a429134) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133291 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3774 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3593 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/100350 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/83187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/57a67ca3-f815-4c73-9ad6-c9f567ed8e1b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134367 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/2726 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117638 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81131 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/9ff66192-1a98-4d97-89dd-9589cf755047) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/2641 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/380 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82119 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/111251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35762 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141562 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3496 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36280 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/108717 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3542 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/3126 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/108934 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27597 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2655 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113967 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56681 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3558 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32383 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3380 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66966 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3580 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3488 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.cpp:
```diff
@@ -28,15 +28,71 @@
 WTF_MAKE_TZONE_ALLOCATED_IMPL(RemoteMediaSessionClientProxy);
 
-RemoteMediaSessionClientProxy(RemoteMediaPlayerState& state, RemoteMediaSessionManagerProxy& manager)
+RemoteMediaSessionClientProxy::RemoteMediaSessionClientProxy(RemoteMediaSessionState& state, RemoteMediaSessionManagerProxy& manager)
```
```suggestion
RemoteMediaSessionClientProxy::RemoteMediaSessionClientProxy(const RemoteMediaSessionState& state, 
```

jyavenard commented on Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.cpp:
```diff
@@ -28,15 +28,71 @@
+{
+}
+
+RemoteMediaSessionClientProxy::~RemoteMediaSessionClientProxy()
```
```suggestion
RemoteMediaSessionClientProxy::~RemoteMediaSessionClientProxy() = default;
```

jyavenard commented on Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.h:
```diff
@@ -1,4 +1,4 @@
-/*
+ /*
```
I don't think that space should be added (here and elsewhere)

jyavenard commented on Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.h:
```diff
@@ -29,43 +29,59 @@
-    void ref() const { RefCounted::ref(); }
-    void deref() const { RefCounted::deref(); }
+    void ref() const final { ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr::ref(); }
+    void deref() const final { ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr::deref(); }
```
```suggestion
    WTF_ABSTRACT_THREAD_SAFE_REF_COUNTED_AND_CAN_MAKE_WEAK_PTR_IMPL;
```

jyavenard commented on Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h:
```diff
@@ -0,0 +1,85 @@
+        return adoptRef(*new RemoteMediaSessionProxy(state, manager));
+    }
+
+    virtual ~RemoteMediaSessionProxy();
```
nit, class is final, it doesn't have to be virtual.

jyavenard commented on Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h:
```diff
@@ -0,0 +1,85 @@
+    void updateState(const RemoteMediaSessionState&);
+
+private:
+    RemoteMediaSessionProxy(RemoteMediaSessionState&, RemoteMediaSessionManagerProxy&);
```
```suggestion
    RemoteMediaSessionProxy(const RemoteMediaSessionState&, RemoteMediaSessionManagerProxy&);
```

jyavenard commented on Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.cpp:
```diff
@@ -101,27 +69,164 @@ RemoteMediaSessionManager::~RemoteMediaSessionManager()
+
+void RemoteMediaSessionManager::updateCachedSessionState(const WebCore::PlatformMediaSessionInterface& session, RemoteMediaSessionState& state)
+{
+    state.state = session.state();
```
Prefers the previously used syntax and it's less verbose.

```suggestion
    state = {
        .state = session.state(),
        .stateToRestore = session.stateToRestore(),
```
....

eric-carlson replied:
I don't think it can be less verbose, it is initializing a `RemoteMediaSessionState` from a `PlatformMediaSessionInterface`

jyavenard commented on Source/WebKit/WebProcess/Media/RemoteMediaSessionState.h:
```diff
@@ -30,24 +30,28 @@
 namespace WebKit {
 
 struct RemoteMediaSessionState {
+    WTF_DEPRECATED_MAKE_STRUCT_FAST_ALLOCATED(RemoteMediaSessionState);
```
Do we need this considering it's deprecated ?

eric-carlson replied:
Good point, switched to `WTF_MAKE_STRUCT_TZONE_ALLOCATED`

jyavenard reviewed (approved):
r+ with nits

eric-carlson committed:
[Site Isolation] Begin filling out and connecting RemoteMediaSessionManager and RemoteMediaSessionManagerProxy
https://bugs.webkit.org/show_bug.cgi?id=302546
rdar://164742719

Reviewed by Jean Yves Avenard.

Beging filling out the implementations of RemoteMediaSessionManager and
RemoteMediaSessionManagerProxy.

No new tests, the remote session manager is disabled by default and is not testable yet.

* Source/WebCore/platform/audio/MediaSessionManagerInterface.cpp:
(WebCore::MediaSessionManagerInterface::addSession):
* Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
(WebCore::MediaSessionManagerInterface::logger const):
* Source/WebCore/platform/audio/PlatformMediaSession.h:
(WebCore::PlatformMediaSession::create): Deleted.
(WebCore::PlatformMediaSession::PlatformMediaSession): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
(WebCore::PlatformMediaSessionInterface::setMediaSessionIdentifier):
(WebCore::PlatformMediaSessionInterface::client const):
(WebCore::PlatformMediaSessionInterface::checkedClient const):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::sessions const):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::updateSessionState):
(WebCore::MediaSessionManagerCocoa::scheduleSessionStatusUpdate):
(WebCore::MediaSessionManagerCocoa::addSession):
(WebCore::MediaSessionManagerCocoa::removeSession):
(WebCore::MediaSessionManagerCocoa::setCurrentSession):
(WebCore::MediaSessionManagerCocoa::addSupportedCommand):
(WebCore::MediaSessionManagerCocoa::removeSupportedCommand):
(WebCore::MediaSessionManagerCocoa::supportedCommands const):
(WebCore::MediaSessionManagerCocoa::updateNowPlayingInfo):
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
(WebCore::MediaSessionManageriOS::logClassName const):
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.cpp:
(WebKit::RemoteMediaSessionClientProxy::RemoteMediaSessionClientProxy):
(WebKit::RemoteMediaSessionClientProxy::resumeAutoplaying):
(WebKit::RemoteMediaSessionClientProxy::mayResumePlayback):
(WebKit::RemoteMediaSessionClientProxy::suspendPlayback):
(WebKit::RemoteMediaSessionClientProxy::didReceiveRemoteControlCommand):
(WebKit::RemoteMediaSessionClientProxy::shouldOverrideBackgroundPlaybackRestriction const):
(WebKit::RemoteMediaSessionClientProxy::setShouldPlayToPlaybackTarget):
(WebKit::RemoteMediaSessionClientProxy::sessionManager const):
(WebKit::RemoteMediaSessionClientProxy): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.h:
(WebKit::RemoteMediaSessionClientProxy::create): Deleted.
(WebKit::RemoteMediaSessionClientProxy::selectBestMediaSession): Deleted.
(WebKit::RemoteMediaSessionClientProxy::protectedLogger const): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.cpp:
(WebKit::RemoteMediaSessionManagerProxy::RemoteMediaSessionManagerProxy):
(WebKit::RemoteMediaSessionManagerProxy::addMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::removeMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::setCurrentMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::updateMediaSessionState):
(WebKit::RemoteMediaSessionManagerProxy::findSession):
(WebKit::RemoteMediaSessionManagerProxy::addSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::removeSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::setCurrentSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::updateSessionState): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.h:
(WebKit::RemoteMediaSessionManagerProxy::process const):
(WebKit::RemoteMediaSessionManagerProxy::ref const): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::deref const): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp:
(WebKit::RemoteMediaSessionProxy::RemoteMediaSessionProxy):
(WebKit::RemoteMediaSessionProxy::~RemoteMediaSessionProxy):
(WebKit::RemoteMediaSessionProxy::updateState):
(WebKit::RemoteMediaSessionProxy::setShouldPlayToPlaybackTarget):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h: Added.
* Source/WebKit/UIProcess/XR/PlatformXRSystem.h:
* Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp:
(WebKit::SystemSettingsManagerProxy::updateFontProperties):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.cpp:
(WebKit::RemoteMediaSessionManager::addSession):
(WebKit::RemoteMediaSessionManager::removeSession):
(WebKit::RemoteMediaSessionManager::setCurrentSession):
(WebKit::RemoteMediaSessionManager::updateSessionState):
(WebKit::RemoteMediaSessionManager::sessionWithIdentifier):
(WebKit::RemoteMediaSessionManager::clientShouldResumeAutoplaying):
(WebKit::RemoteMediaSessionManager::clientMayResumePlayback):
(WebKit::RemoteMediaSessionManager::clientShouldSuspendPlayback):
(WebKit::RemoteMediaSessionManager::clientSetShouldPlayToPlaybackTarget):
(WebKit::RemoteMediaSessionManager::clientDidReceiveRemoteControlCommand):
(WebKit::RemoteMediaSessionManager::currentSessionState):
(WebKit::RemoteMediaSessionManager::updateCachedSessionState):
(WebKit::RemoteMediaSessionManager::fullSessionState):
(WebKit::RemoteMediaSessionManager::sharedPreferencesForWebProcess const):
(WebKit::platformSessionState): Deleted.
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.h:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.messages.in:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.h:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.serialization.in:

Canonical link: https://commits.webkit.org/303123@main
